File: srv_pkg/scripts/service_client.py
#!/usr/bin/env python

from __future__ import print_function

import sys
import rospy
from srv_pkg.srv import *

def add_two_ints_client(x, y):
    rospy.wait_for_service('add_two_ints')
    try:
        add_two_ints = rospy.ServiceProxy('add_two_ints', test_srv)
        resp1 = add_two_ints(x, y)
        return resp1.sum
    except rospy.ServiceException as e:
        print("Service call failed: %s"%e)
        

if __name__ == "__main__":
    x = 1
    y = 2
    print("Requesting %s+%s"%(x, y))
    print("%s + %s = %s"%(x, y, add_two_ints_client(x, y)))

File: srv_pkg/src/service_client.cpp
#include "ros/ros.h"
#include "srv_pkg/test_srv.h"

int main(int argc, char **argv)
{
  ros::init(argc, argv, "add_two_ints_client");

  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<srv_pkg::test_srv>("add_two_ints");
  srv_pkg::test_srv srv;
  
  srv.request.a = 1;
  srv.request.b = 2;
  if (client.call(srv))
  {
    ROS_INFO("Sum: %ld", (long int)srv.response.sum);
  }
  else
  {
    ROS_ERROR("Failed to call service add_two_ints");
    return 1;
  }

  return 0;
}